lib/Dialect/TTIR/Transforms/Passes.cpp

```diff
@@ -681,7 +694,9 @@ class TTIRLayoutDPSOperandsRewriter
 
     assert(op->template hasTrait<TTIROp::Trait>());
     bool modified = false;
-    for (auto &operand : op->getOpOperands()) {
+    for (int64_t operandIndex = 0, numOperands = op->getNumOperands();
+         operandIndex < numOperands; ++operandIndex) {
```

Comment:
Can't this be `for (int64_t operandIndex = 0; operandIndex < op->getNumOperands(); ++operandIndex)`?

I guess it's a perf optimization, do we know if it's worth it though?

Reply:
Sure.